packages/components/src/utils/button-handling.ts

```diff
@@ -1,19 +1,25 @@
 import type { ButtonType } from '../types';
-import { getClosestHTMLElement } from './dom';
+import { getClosestHTMLElement, setAttributes } from './dom';
 
 export const improveButtonHandlingForCustomElement = (
   element: HTMLElement,
   getType: () => ButtonType,
-  getDisabled: () => boolean
+  getDisabled: () => boolean,
+  getName?: () => string | undefined,
+  getValue?: () => string | undefined
 ): void => {
-  element.addEventListener('click', (event) => handleButtonEvent(event, element, getType, getDisabled));
+  element.addEventListener('click', (event) =>
+    handleButtonEvent(event, element, getType, getDisabled, getName, getValue)
+  );
 };
 
 export const handleButtonEvent = (
   event: MouseEvent | KeyboardEvent,
   element: HTMLElement,
   getType: () => ButtonType,
-  getDisabled: () => boolean
+  getDisabled: () => boolean,
+  getName?: () => string | undefined,
```

Comment:
Why optional? Would be better typed and safer if it was always necessary.
Then we wouldn't need optional chaining down below, too.

Reply:
Since it is used in the `multi-select` and `text-field-wrapper` without those params